### PR TITLE
Feature/rdm 11131 audit logging

### PR DESCRIPTION
**Pull request by sem-hmcts**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RDM-11131

### Change description ###
Create and write audit logs to the standard output per the above Jira specification.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/reform/ccd/documentam/auditlog/aop/ExpressionEvaluatorTest.java [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+
+        private void hello(String foo, Boolean flag) {
+        }
+
+        private void hello(String foo, SampleBean sampleBean) {
+        }
+    }
+
+    // @Data
+    // @AllArgsConstructor
```
Comment (sullivangareth) [inline src/test/java/uk/gov/hmcts/reform/ccd/documentam/auditlog/aop/ExpressionEvaluatorTest.java:145]: Remove these commented out lines

File: src/main/resources/application.yaml [hunk 2]
```diff
@@ -1,6 +1,14 @@
 server:
   port: ${PORT:4455}
 
+audit:
+  log:
+    enabled: true
+    ignore:
+      # http statuses with comma separated to ignore audit logs
+      statuses:
```
Comment (sullivangareth) [inline src/main/resources/application.yaml:9]: What is the scenario where we would have audit logs disabled ?   Do we need to override any of this audit configuration in the charts files ?

**Review by sullivangareth — DISMISSED** [review]

File: src/main/resources/application.yaml (hunk 2 above)
Comment (sem-hmcts) [reply src/main/resources/application.yaml:9]: @sullivangareth 
going by the cpo-case-payment-orders-api example, which this implementation follows, the audit log is always enabled.  The Jira ticket never actually asked for the audit logging feature to be configurable this way.  I'm assuming this was requested by some other requirement.  Shall we take it out altogether if it's confusing?

File: src/test/java/uk/gov/hmcts/reform/ccd/documentam/auditlog/aop/ExpressionEvaluatorTest.java (hunk 1 above)
Comment (sem-hmcts) [reply src/test/java/uk/gov/hmcts/reform/ccd/documentam/auditlog/aop/ExpressionEvaluatorTest.java]: Done

File: src/main/resources/application.yaml (hunk 2 above)
Comment (smathangi) [reply src/main/resources/application.yaml:9]: I think it's already there https://github.com/hmcts/ccd-case-document-am-api/pull/412/files#diff-2109c901004111239683f91813f776c081fc8ec9b94eda17d5dc3142cdc289c1R22
The default in the application.yaml is true. So, it's configurable. 
Whenever we want to disable we can always add an environment variable in the related flux config.

File: src/main/java/uk/gov/hmcts/reform/ccd/documentam/auditlog/LogAudit.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+    /**
+     * Expression to parse when loading the Case ID from a string value.
+     */
+    String caseId() default "";
+
+    /**
+     * Expression to parse when loading a list of Case IDs.
+     */
+    String caseIds() default "";
```
Comment (smathangi) [inline src/main/java/uk/gov/hmcts/reform/ccd/documentam/auditlog/LogAudit.java:35]: i think we don't have list of caseIds use case here. May be remove from annotation and aspect as well?

Comment (sem-hmcts) [reply src/main/java/uk/gov/hmcts/reform/ccd/documentam/auditlog/LogAudit.java]: Fixed

**Review by smathangi — APPROVED** [review]
LGTM. Thanks @sem-hmcts

**Review by sullivangareth — APPROVED** [review]

**Pull request merged**